src/index.ts

```diff
@@ -7,7 +7,7 @@ type UserConfigs = Parameters<typeof antfu>[1];
 type ESLintConfig = ReturnType<typeof antfu>;
 
 // eslint-disable-next-line antfu/no-top-level-await
-const tsconfigPath = await resolveTSConfig();
+const tsconfigPath = await resolveTSConfig().then(path => path).catch(() => undefined);
```

Comment:
```suggestion
const tsconfigPath = await resolveTSConfig().catch(() => undefined);
```

Is there a reason you need then?

Reply:
both works totally fine, but which is more readable for you? For me, first one.

Reply:
I think it's confusing because it makes it seem like it means something.
I don't think much of it because it's clear that it doesn't really mean anything.

My preference is the latter, but it doesn't matter.

Reply:
Maybe it was difficult to read because I wrote it in one-liners!

If you wrote the latter code I would not reject it.
It's hard to discuss too much about coding styles that lint doesn't detect, so let's stop at this point.
Your opinion is plausible.

Reply:
To go a little deeper
Promises are moand like objects.
That's why then and catch are equivalent to conditional maps.

Math is all about stripping out unnecessary formulas.

Of course, you know better than me
In that sense, it's not a favorite.

---

Of course, I think that code is saying, “I'm going to give you the value when it's not a problem.”
So that's fine.

Reply:
you are so smart.
And I actually I wrote those kind of codes a lot while ago.
Not only this (actually this is not dangerous), but I used a lot of `!!` of falsy variables in typescript and python a lot.

But that made it hard to understand what the code meant afterwards, which was painful. So I try to write the code as clearly as possible, without abbreviations, so that it is easy to understand what the code is explicitly doing.
So I don't use falsy boolean as much as possible, and I always write ‘then’ in a promise.
This is completely my preference and I am not denying your preference.
In fact, it looks good in some cases for a `promise` to omit `then`. Thank you.

Reply:
Also, please distinguish between areas that require explanation and those that don’t. These are the places where we shouldn’t spend too much time.

Reply:
I think it's just a process of coming to terms with each other.
Of course, for the sake of speed and to avoid time-consumption, we're going to use the
I would approve first.

Because there's no reason why this should be a PR blocker.

If you still feel like it's taking up your time, well...

---

Certainly, I've asked a lot of questions to clarify my doubts in a lot of places.
That's definitely an area where time shouldn't be spent.
That's why I put “just curious” at the end or at the beginning.

I think that probably didn't convey what I was trying to say, so I think that's a lot of wasted time.
For example, I've gotten the wrong answer to a question and had to go through the process of adjusting it.

The obvious solution is for me to learn English more deeply (I'm working on it...).
Otherwise, um...
Maybe I should give Nit a try?

Reply:
Yeah let's add some small words at the beginning like nits question like that. I'll do it! Thanks as always

Reply:
Also my attitude is not perfect yesterday. I'm sorry

Reply:
I'm fine, I just realized it wasn't perfect